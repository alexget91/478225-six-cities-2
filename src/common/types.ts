import {PlaceType} from "./constants";

type Host = {
  id: number,
  name: string,
  avatar_url?: string,
  is_pro?: boolean,
}

type User = {
  email: string,
} & Host;

type Location = {
  latitude: number,
  longitude: number,
  zoom?: number,
}

type City = {
  name: string,
  location: Location,
};

type PlaceCard = {
  id: number,
  city?: City,
  preview_image?: string,
  images?: Array<string>,
  title: string,
  is_premium?: boolean,
  is_favorite?: boolean,
  rating?: number,
  type: PlaceType,
  bedrooms?: number,
  max_adults?: number,
  price: number,
  goods?: Array<string>,
  host?: Host,
  description?: string,
  location?: Location,
};

type PlaceList = Array<PlaceCard>;

type ReviewItem = {
  id: number,
  user: Host,
  rating: number,
  comment: string,
  date: string,
};

type ReviewsList = Array<ReviewItem>;

type PinCoordinates = [number, number];

type Pin = {
  coords: PinCoordinates,
  isActive?: boolean,
}

type AnyProps = {
  [propName: string]: any
};

export {User, City, PlaceCard, PlaceList, ReviewItem, ReviewsList, PinCoordinates, Pin, AnyProps};
