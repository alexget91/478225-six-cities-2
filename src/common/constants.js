const PageType = {
  FAVORITES: `FAVORITES`,
  FAVORITES_EMPTY: `FAVORITES_EMPTY`,
  LOGIN: `LOGIN`,
  MAIN: `MAIN`,
  MAIN_EMPTY: `MAIN_EMPTY`,
  OFFER: `OFFER`,
};

const PlacesListView = {
  OFFER: `OFFER`,
  LIST: `LIST`,
  FAVORITES: `FAVORITES`,
};

const sortingOptions = {
  popular: `popular`,
  toHigh: `to-high`,
  toLow: `to-low`,
  topRated: `top-rated`,
};

const FormSendingStatus = {
  READY: `READY`,
  SUCCESS: `SUCCESS`,
  SENDING: `SENDING`,
};

export {PageType, PlacesListView, sortingOptions, FormSendingStatus};
