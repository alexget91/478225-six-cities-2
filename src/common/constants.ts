enum PageType {
  FAVORITES = `FAVORITES`,
  FAVORITES_EMPTY = `FAVORITES_EMPTY`,
  LOGIN = `LOGIN`,
  MAIN = `MAIN`,
  MAIN_EMPTY = `MAIN_EMPTY`,
  OFFER = `OFFER`,
}

enum PlaceType {
  APARTMENT = `apartment`,
  ROOM = `room`,
  HOUSE = `house`,
  HOTEL = `hotel`,
}

enum PlacesListView {
  OFFER = `OFFER`,
  LIST = `LIST`,
  FAVORITES = `FAVORITES`,
}

enum SortingOption {
  POPULAR = `popular`,
  TO_HIGH = `to-high`,
  TO_LOW = `to-low`,
  TOP_RATED = `top-rated`,
}

enum FormSendingStatus {
  READY = `READY`,
  SUCCESS = `SUCCESS`,
  SENDING = `SENDING`,
}

export {PageType, PlaceType, PlacesListView, SortingOption, FormSendingStatus};
