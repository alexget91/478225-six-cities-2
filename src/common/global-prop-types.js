import PropTypes from "prop-types";

const placeCard = {
  id: PropTypes.number.isRequired,
  previewImage: PropTypes.string,
  images: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string.isRequired,
  isPremium: PropTypes.bool,
  isFavorite: PropTypes.bool,
  rating: PropTypes.number,
  type: PropTypes.oneOf([`apartment`, `room`, `house`, `hotel`]).isRequired,
  bedrooms: PropTypes.number,
  maxAdults: PropTypes.number,
  priceByNight: PropTypes.number.isRequired,
  goods: PropTypes.arrayOf(PropTypes.string),
  host: PropTypes.exact({
    isPro: PropTypes.bool,
    name: PropTypes.string,
    avatarUrl: PropTypes.string
  }),
  description: PropTypes.string,
};

const placeList = PropTypes.arrayOf(PropTypes.exact(placeCard));

export {placeCard, placeList};
