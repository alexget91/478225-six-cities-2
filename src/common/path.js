const Path = {
  INDEX: `/`,
  OFFER: `/offer`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
};

export default Path;
