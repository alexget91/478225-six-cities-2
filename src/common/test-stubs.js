const createMapBlock = () => {
  const div = global.document.createElement(`div`);
  div.setAttribute(`id`, `map`);
  global.document.body.appendChild(div);
};

const getMockOffer = (id, cityName) => {
  return {
    id,
    city: {
      name: cityName,
      location: {
        latitude: 0,
        longitude: 0,
      }
    },
    priceByNight: 0,
    title: `Title${id}`,
    type: `apartment`,
    location: {
      latitude: 0,
      longitude: 0,
    }
  };
};

export {createMapBlock, getMockOffer};
