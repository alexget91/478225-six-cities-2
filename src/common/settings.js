import {sortingOptions} from "./constants";

const Settings = {
  API_TIMEOUT: 5000,
  DEFAULT_SORT: sortingOptions.popular,
  COMMENT_LENGTH_MIN: 50,
  COMMENT_LENGTH_MAX: 300,
};

export default Settings;
