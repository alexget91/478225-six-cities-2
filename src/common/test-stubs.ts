import {PlaceType} from "./constants";
import {PlaceCard} from "./types";

const createMapBlock = (): void => {
  const div: HTMLDivElement = (global as any).document.createElement(`div`);
  div.setAttribute(`id`, `map`);
  (global as any).document.body.appendChild(div);
};

const getMockOffer = (id: number, cityName: string, isFavorite?: boolean): PlaceCard => ({
  id,
  city: {
    name: cityName,
    location: {
      latitude: 0,
      longitude: 0,
    },
  },
  is_favorite: isFavorite || false,
  price: 0,
  title: `Title${id}`,
  type: PlaceType.APARTMENT,
  location: {
    latitude: 0,
    longitude: 0,
  },
});

export {
  createMapBlock,
  getMockOffer,
};
