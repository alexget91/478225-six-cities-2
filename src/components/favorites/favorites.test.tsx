import * as React from "react";
import * as renderer from "react-test-renderer";
import Favorites from "./favorites";
import {getMockOfferTransformed} from "../../common/test-stubs";
import {MemoryRouter} from "react-router-dom";

describe(`Favorites correctly renders after relaunch`, () => {
  it(`With content`, () => {
    const tree = renderer
      .create(<MemoryRouter><Favorites
        isLoaded={true}
        offers={{
          city: [getMockOfferTransformed(1, `city`, true)],
        }}
      /></MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When empty`, () => {
    const tree = renderer
      .create(<Favorites
        isLoaded={true}
        offers={{}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When is not loaded`, () => {
    const tree = renderer
      .create(<Favorites/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
