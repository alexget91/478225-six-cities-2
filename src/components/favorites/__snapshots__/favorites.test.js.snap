// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites correctly renders after relaunch When empty 1`] = `
<div
  className="page__favorites-container container"
>
  <section
    className="favorites favorites--empty"
  >
    <h1
      className="visually-hidden"
    >
      Favorites (empty)
    </h1>
    <div
      className="favorites__status-wrapper"
    >
      <b
        className="favorites__status"
      >
        Nothing yet saved.
      </b>
      <p
        className="favorites__status-description"
      >
        Save properties to narrow down search or plan yor future trips.
      </p>
    </div>
  </section>
</div>
`;

exports[`Favorites correctly renders after relaunch When is not loaded 1`] = `"Loading..."`;

exports[`Favorites correctly renders after relaunch With content 1`] = `
<div
  className="page__favorites-container container"
>
  <section
    className="favorites"
  >
    <h1
      className="favorites__title"
    >
      Saved listing
    </h1>
    <ul
      className="favorites__list"
    >
      <li
        className="favorites__locations-items"
      >
        <div
          className="favorites__locations locations locations--current"
        >
          <div
            className="locations__item"
          >
            <a
              className="locations__item-link"
              href="#"
            >
              <span>
                city
              </span>
            </a>
          </div>
        </div>
        <div
          className="favorites__places"
        >
          <article
            className="favorites__card place-card"
            data-id={1}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            
            <div
              className="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                className="js-detail-link"
                href="/offer/1"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={100}
                  width={150}
                />
              </a>
            </div>
            <div
              className="favorites__card-info place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    0
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                     / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button js-favorites-link"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  className="js-detail-link"
                  href="/offer/1"
                  onClick={[Function]}
                >
                  Title1
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
